urnai/actions/chain_of_actions_base.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABC
+
+from action_base import ActionBase
+
+from urnai.utils.returns import List
+
+
+class ChainOfActionsBase(ABC):
+
+	"""This class works as a sequence of actions"""
+	
+	action_list: List[ActionBase]
+	
+	def __init__(self):
+		self.action_list = []
+	
+	def next(self):
+		# Executing the next action in the sequence
+		
+		if self.get_len() == 0: 
+			return None
+	
+		first_action = self.action_list[0]
+		
+		if first_action.is_complete():
+			self.action_list.pop(0)
+			self.next()
```

Comment:
I don't agree with this approach. The way it is, I wouldn't be able to reuse an `ChainOfActionsBase` object, because the list of actions would be empty at the end. I think we could use an index to just select the actions, which would allow us to reuse the object in several executions.